src/cheshire/core.clj

```diff
@@ -24,9 +49,12 @@
          generator (.createGenerator
                     ^JsonFactory (or factory/*json-factory*
                                      factory/json-factory)
-                    ^Writer sw)]
-     (when (:pretty opt-map)
+                    ^Writer sw)
+         print-pretty (:pretty opt-map)]
+     (when (= true print-pretty)
        (.useDefaultPrettyPrinter generator))
+     (when (map? print-pretty)
```

Comment:
Seems like it'd be better to be able to re-use the pretty printer, so each new generation didn't have to create more objects